File: 1915-check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        int n = s1.length();

        int count_s1[26] = {};
        int count_s2[26] = {};

        for(int i = 0; i < n; ++i) {
            count_s1[s1[i] - 'a']++;
            count_s2[s2[i] - 'a']++;
        }

        for(int i = 0; i < 26; ++i)
            if(count_s1[i] != count_s2[i])
                return false;

        int diff = 0;
        for(int i = 0; i < n; ++i) 
            diff += s1[i] != s2[i] ? 1 : 0;

        return diff <= 2;
    }
};

File: 1915-check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        if (s1.equals(s2)) return true;
        List<Integer> swapIndex = new ArrayList<>();
        for (int i= 0; i<s1.length();i++) {
            if (s1.charAt(i) != s2.charAt(i)) {
                swapIndex.add(i);
            }
        }
        if (swapIndex.size() == 2) {
            int i =swapIndex.get(0), j =swapIndex.get(1);
            return (s1.charAt(i) == s2.charAt(j) && s1.charAt(j) == s2.charAt(i));
        }return false;
    }
}